### DISCUSSION TITLE
[question] Possible to include some JS libs?

### DISCUSSION THREAD

**Discussion by globules-io** [c-smile/sciter-js-sdk#37] [General]
Just discovering this project and I was wondering if it's possible to include any js lib that we want? If yes, how? Thanks

**Comment by c-smile** [comment]
Sciter.JS exists in three different forms:

1. As scapp.exe - that is standalone executable that includes full runtime of HTML/CSS/JS. Intended usage: 
    *  "as it is" - to run HTML applications from files on hard drive, or
    *  "compiled" applications a la [Sciter.Quark](https://quark.sciter.com/)  
2. As sciter.dll - that is embeddable version of the engine, your native application can use it as UI and logic layer. Examples of such usage are usciter.exe and [inspector.exe, (kind of external Firebug)](https://sciter.com/sciter-js-inspector-debugger-preactjs/) applications from SDK.
   
   Note that usciter.exe and inspector.exe include all HTML/CSS/JS resources compiled into them as resources. 

   [Here is](https://github.com/c-smile/sciter-js-sdk/tree/main/demos/inspector/res), for example, folder that contains all HTML/CCS/JS that constitute the inspector. Content of this folder gets compressed and packaged into [a BLOB](https://github.com/c-smile/sciter-js-sdk/blob/main/demos/inspector/resources.cpp) that is compiled into exe. Packaging is done by [packfolder utility](https://github.com/c-smile/sciter-js-sdk/blob/main/demos/inspector/pack-resources.bat)  from the SDK. 

   And of course you can include into that folder (and so your binary) any JS library (and HTML/CSS) you want.

3. And the third form is use of sciter.lib - static library that can be compiled into your executable statically. So you may have monolithic executable that includes HTML/CSS/JS runtime, your resources and may be some other native libraries like SQLite for example.

**Reply by globules-io** [reply]
Thanks for taking the time for this lengthy reply! Much appreciated!
